### ISSUE TITLE
Feature Request: Create process diagrams with Mermaid CLI

### ISSUE THREAD

**Issue by epompeii** [jhpyle/docassemble#93]
In multi-user and multi-session interviews, it would be great to visually represent to the user both a process overview and the user's position within that process. The Section Navigation Bar and the Progress Bar handle some aspects of this. However, there are some cases where it would be useful to be able to give 10,000 foot view of the process at a glance.

[Mermaid](https://mermaidjs.github.io/) is to process diagrams what markdown is to HTML. [Mermaid CLI](https://mermaidjs.github.io/mermaidCLI.html) is just a node package, so it should be pretty easy to add in, dependency wise.

There's also a [pandoc filter](https://github.com/raghur/mermaid-filter) for it, which should allow one to incorporate the diagrams in md -> pdf, docx, etc.

As far as diagrams go, mermaid supports creating sequence diagrams, gantt charts, and (heresy I know) flowcharts.

A bonus for documentation purposes, there is also [a ruby gem for Jekyll](https://rubygems.org/gems/jekyll-mermaid/versions/1.0.0).

**Comment by jhpyle** [comment]
What do you envision this integration looking like?  I am thinking that there would be a special block type, similar to `template`, that looks like this:
```
mermaid: my_diagram
content: |
  graph TD;
      A-->B;
      A-->C;
      B-->D;
      C-->D;
```

Then, the interview author would be able to write something like:
```
question: Here is where we are.
subquestion: |
  ${ my_diagram }
field: user_has_seen_diagram
```

**Comment by epompeii** [comment]
I think that's a great idea!
My hope was to also be able to include them in the Help section as well, which this also allows.

This may not be possible, but I was also hoping that one would be able to include mako in the mermaid block. That way one could reuse the same mermaid block throughout. For example, highlighting the current stage in the process:

```
mermaid: my_diagram
content: |
  graph TD;
      A-->B |
      %if i_am_at_point_b == true
      fill:#f9f,stroke:#333,stroke-width:4px;
      % else:
      fill:#ccf,stroke:#f66,stroke-width:2px,stroke-dasharray: 5, 5;
```

If this does work, would the user have to use `sets`? I'm still a little unclear on when `docassemble` needs that helping hand of `sets`.

**Comment by jpylephilalegal** [comment]
Yes, it would be possible to use Mako.  And this block would not need `sets` because the variable it defines is already apparent from the `mermaid: my_diagram` line.  A `sets` directive is necessary only when you are [generating fields with code](https://docassemble.org/docs/fields.html#fields%20code) or you set `scan for variables` to `False`.
